transcendence-app/src/user/api42.service.ts

```diff
@@ -25,6 +25,15 @@ export class Api42Service {
     return data;
   }
 
+  downloadUserAvatar(accessToken: string, url: string) {
+    return lastValueFrom(
```

Comment:
> Converts an observable to a promise by subscribing to the observable, waiting for it to complete, and resolving the returned promise with the last value from the observed stream.
If the observable stream completes before any values were emitted, the returned promise will reject with EmptyError or will resolve with the default value if a default was specified.
If the observable stream emits an error, the returned promise will reject with that error.
WARNING: Only use this with observables you know will complete. If the source observable does not complete, you will end up with a promise that is hung up, and potentially all of the state of an async function hanging out in memory. To avoid this situation, look into adding something like timeout, take, takeWhile, or takeUntil amongst others.

Specially:

> **WARNING: Only use this with observables you know will complete**. If the source observable does not complete, you will end up with a promise that is hung up, and potentially all of the state of an async function hanging out in memory

Are we sure lastValueFrom is the best option for this?

>  To avoid this situation, look into adding something like timeout, take, takeWhile, or takeUntil amongst others.

Do we know a request to the 42Api will complete?

Reply:
Good catch!
I've added a timeout to the HttpModule. It should be good now.
Should we make it an environment variable or keep it as a constant?